components/rmrk/Gallery/CollectionPriceChart.vue

```diff
@@ -1,160 +1,180 @@
+import Chart from 'chart.js/auto';
+import 'chartjs-adapter-luxon';
+import zoomPlugin from 'chartjs-plugin-zoom';
 
-import { LegendComponent } from 'echarts/components'
+Chart.register(zoomPlugin);
+type PriceDataArray = [Date, string | number];
 
-ECharts.use([GridComponent, LineChart, CanvasRenderer, UniversalTransition, TooltipComponent, DataZoomComponent, LegendComponent])
-
-const components = {
-}
+const components = {};
 
 @Component({ components })
 export default class PriceChart extends Vue {
   @Prop() public priceData!: any[];
-  // @Prop() public eventData!: Date[];
 
   protected chartOptionsLine: any = {};
-  protected Chart!: ECharts.ECharts;
+  protected Chart!: Chart<'line', any, unknown>;
+
+  protected resetZoom(): void {
+    this.Chart.resetZoom();
+  }
 
   protected onWindowResize() {
     if (this.Chart) {
-      this.Chart.resize({ width: 'auto', height: 400 })
+      this.Chart.resize();
     }
   }
 
+  protected onCanvasMouseDown(): void {
+    document!.getElementById('collectionPriceChart')!.style.cursor = 'grabbing';
+  }
+
+  protected onCanvasMouseUp(): void {
+    document!.getElementById('collectionPriceChart')!.style.cursor = 'auto';
+  }
+
+  protected onCanvasMouseLeave(): void {
+    document!.getElementById('collectionPriceChart')!.style.cursor = 'auto';
+  }
+
   public async created() {
-    window.addEventListener('resize', this.onWindowResize)
+    window.addEventListener('resize', this.onWindowResize);
   }
 
   public async mounted() {
-    this.priceChart()
+    this.priceChart();
   }
 
   protected priceChart() {
-
-    this.Chart = ECharts.init(this.$refs.chart as HTMLElement)
-    this.Chart.setOption({
-      tooltip: {
-        trigger: 'item',
-        formatter: (params: { data: string[], seriesName: string }) => {
-          const date = this.parseDate(
-            params.data[0] as unknown as Date
-          )
-          const price = params.data[1] + ' KSM'
-
-          return '<center>' + date + '<br>' + price + '</center>'
-        },
-        backgroundColor: '#363636',
-        textStyle: {
-          color: '#fff',
-          fontFamily: 'Fira Code',
+    if (this.priceData.length) {
+      const ctx = (
+        document?.getElementById('collectionPriceChart') as HTMLCanvasElement
+      )?.getContext('2d')!;
+      const chart = new Chart(ctx, {
+        type: 'line',
+        data: {
+          datasets: [
+            {
+              label: 'Floor Price',
+              data: this.priceData[0].map((data: PriceDataArray) => ({
+                x: data[0],
+                y: data[1],
+              })),
+              borderColor: '#d32e79',
+              tension: 0.3,
+              pointBackgroundColor: 'white',
+              pointBorderColor: 'blue',
+              pointRadius: 4,
+              pointHoverRadius: 6,
+            },
+            {
+              label: 'Sold NFT Price',
+              data: this.priceData[1].map((data: PriceDataArray) => ({
```

Comment:
I would move this that arrow function into the utils

Reply:
On it